doc/advpng.1
.TH "AdvanceCOMP PNG Compression Utility" 1
.SH NAME
advpng \- AdvanceCOMP PNG Compression Utility
.SH SYNOPSIS 
advpng [\-l, \-\-list] [\-z, \-\-recompress] [\-0, \-\-shrink\-0]
.PD 0
.PP
.PD
	[\-1, \-\-shrink\-fast] [\-2, \-\-shrink\-normal [\-3, \-\-shrink\-extra]
.PD 0
.PP
.PD
	[\-4, \-\-shrink\-insane] [\-i, \-\-iter N]
.PD 0
.PP
.PD
	[\-f, \-\-force] [\-q, \-\-quiet]
.PD 0
.PP
.PD
	[\-h, \-\-help] [\-V, \-\-version] FILES...
.PD 0
.PP
.PD
.SH DESCRIPTION 
The main purpose of this utility is to recompress png
files to get the smallest possible size.
.PP
Please note that this utility is not able to read
a generic file. It\'s granted to be able to read only
the files generated by the AdvanceMAME emulator.
.PP
To compress the files this utility uses the following
strategies:
.PD 0
.IP \(bu
Remove all ancillary chunks.
.IP \(bu
Concatenate all the IDAT chunks.
.IP \(bu
Use the 7zip Deflate implementation.
.PD
.SH OPTIONS 
.TP
.B \-l, \-\-list FILES...
List the content of the specified files.
.TP
.B \-z, \-\-recompress FILES...
Recompress the specified files. If the \-1, \-2, \-3
options are specified it\'s used the smallest file
choice from the previous compressed data and the
new compression. If the \-0 option is specified the
file is always rewritten without any compression.
.TP
.B \-0, \-\-shrink\-store
Disable the compression. The file is
only stored and not compressed. The file is always
rewritten also if it\'s bigger.
.TP
.B \-1, \-\-shrink\-fast
Set the compression level to \[dq]fast\[dq] using the zlib
compressor.
.TP
.B \-2, \-\-shrink\-normal
Set the compression level to \[dq]normal\[dq] using the libdeflate
compressor.
This is the default level of compression.
.TP
.B \-3, \-\-shrink\-extra
Set the compression level to \[dq]extra\[dq] using the 7z
compressor.
You can define the compressor iterations with
the \-i, \-\-iter option.
.TP
.B \-4, \-\-shrink\-insane
Set the compression level to \[dq]insane\[dq] using the zopfli
compressor.
You can define the compressor iterations with
the \-i, \-\-iter option.
.TP
.B \-i, \-\-iter N
Define an additional numer of iterations for the 7z and zopfli
compressors for modes \-3 and \-4.
More iterations usually give a better compression, but may
require a lot more time.
Try for example with 10, 15, 20, and so on.
.TP
.B \-f, \-\-force
Force the use of the new file also if it\'s bigger.
.SH COPYRIGHT 
This file is Copyright (C) 2002 Andrea Mazzoleni, Filipe Estima
.SH SEE ALSO 
advzip(1), advmng(1), advdef(1)


HISTORY
                            =======================
                            History For AdvanceCOMP
                            =======================


ADVANCECOMP VERSION 1.21 2016/09
================================

* Update to latest zopfli.
* Added libdeflate support. It's the new default,
  enabled using -2, --shrink-normal.


ADVANCECOMP VERSION 1.20 2015/08
================================

* Fixed .gz recompression of data bigger than 400 MB.
* Fixed .gz recompression with -0 and -1 options.
* Updated to the latest zopfli [Aaron Kaluszka].
* Preserve the EFS flag in .zip files [Jason Penney].


ADVANCECOMP VERSION 1.19 2014/03
================================

* Limited the iterations number to 255 with the -3 option.
* Added a new -k, --keep-file-time option to advzip [Marco Banfi].
* Made the Windows binary again compatible with old 686 processors.


ADVANCECOMP VERSION 1.18 2013/11
================================

* Added build support for new powerpc architectures.
* Fixed build with BZIP.


ADVANCECOMP VERSION 1.17 2013/03
================================

* Changed to GPL3.
* Added zopfli support. It's enabled using -4, --shrink-insane.
  You can control the number of iterations with the
  new -i, --iter option.
  Thanks to Darik Horn for the patches.


ADVANCECOMP VERSION 1.16 2013/03
================================

* Allowed recompression with files with zero length.
* Some other minor bugfixes.
* Now in git repository.


ADVANCECOMP VERSION 1.15 2005/10
================================

* Fixed the date displayed with the -l advzip command.
* The advdef utility now detects the file type from the file
  header instead of using the extension.
* Fixed a lot of bugs in the 64 bits platforms.


ADVANCECOMP VERSION 1.14 2005/02
================================

* Relaxed a consistency check for the local header in zip files.
  The crc and size entries are allowed to contain the real
  value also if a data descriptor is present.
* Fixed the conversion of RGB images with less than 256 color
  with transparency data.
* Minor fixes at the documentation.


ADVANCECOMP VERSION 1.13 2004/11
================================

* Added support for .svgz files at advdef [rener].
* Fixed the 8 bit color reduction of 32 bit png files.


ADVANCECOMP VERSION 1.12 2004/09
================================

* Fixed a compilation problem with gcc 3.4.


ADVANCECOMP VERSION 1.11 2004/07
================================

* Fixed a FreeBSD libc compatibility issue recompressing
  .gz files [Radim Kolar].
* Fixed a segmentation fault when some invalid
  compressed .zip files are tested.


ADVANCECOMP VERSION 1.10 2004/04
================================

* Added support for alpha channel and the new -n option
  at advmng.
* Fixed the uncompressing error "Invalid compressed data in ..."
  with some GZIP files [Christian Lestrade].


ADVANCECOMP VERSION 1.9 2003/11
===============================

* Added support for .tgz files at advdef.
* Added the -a option at advmng to create .mng files from
  a sequence of .png files.


ADVANCECOMP VERSION 1.8 2003/10
===============================

* Added support for .gz files at advdef.
* Fixed support for .png files with splitted IDATs.


ADVANCECOMP VERSION 1.7 2003/08
===============================

* Fixed a Segmentation Fault bug on the advmng -r option on
  the latest gcc.
* Better and faster (MMX) move recognition in the advmng scroll
  compression.
* The frame reduction of the advmng utility is now done only if possible.
  The compression process never fails.
* Added a new -S (--scroll-square) option at advmng.
* Added a new -v (--verbose) option at advmng to show the
  compression status.
* Changed the internal ID for the bzip2 and LZMA compression.
  The bzip2 ID is now compatible with the PKWARE specification.
* Added support for RGB images with alpha channel at the advpng utility.
* Updated with automake 1.7.6.


ADVANCECOMP VERSION 1.6 2003/05
===============================

* Added the `-x' option at the advmng utility to export .png files
  from a .mng clip. Useful to compress it in an MPEG file.
* Fixed the support for zips with additional data descriptors.
* Updated with autoconf 2.57 and automake 1.7.4.
* Some fixes for the gcc 3.3 compiler.


ADVANCECOMP VERSION 1.5 2003/01
===============================

* Splitted from AdvanceSCAN
* Added the `advdef' compression utility.


ADVANCESCAN VERSION 1.4 2002/12
===============================

* Fixed a bug in the advmng utility when it was called with
  more than one file in the command line. The program
  was incorrectly adding a PLTE chunk at rgb images.


ADVANCESCAN VERSION 1.3 2002/11
===============================

* Added the support for the transparency tRNS chunk at the
  advpng utility.
* Upgraded at the latest Advance Library.
* Fixes at the docs. [Filipe Estima]
* Minor changes at the autoconf/automake scripts.


ADVANCESCAN VERSION 1.2 2002/08
===============================

* Added the advpng utility to compress the PNG files.
* Added the advmng utility to compress the MNG files.
* Added a Windows version.
* Other minor fixes.


ADVANCESCAN VERSION 1.1 2002/06
===============================

* Fixed an infinite loop bug testing some small damaged zips.
* Removed some warning compiling with gcc 3.1.


ADVANCESCAN VERSION 1.0 2002/05
===============================

* First public release.
* Fixed the compression percentage computation on big files.
* Added the --pedantic option at the advzip utility. These
  tests are only done if requested.


ADVANCESCAN VERSION 0.6-BETA 2002/05
====================================

* Added the AdvanceZIP utility.


doc/advpng.d
Name{number}
	advpng - AdvanceCOMP PNG Compression Utility

Synopsis
	:advpng [-l, --list] [-z, --recompress] [-0, --shrink-0]
	:	[-1, --shrink-fast] [-2, --shrink-normal [-3, --shrink-extra]
	:	[-4, --shrink-insane] [-i, --iter N]
	:	[-f, --force] [-q, --quiet]
	:	[-h, --help] [-V, --version] FILES...

Description
	The main purpose of this utility is to recompress png
	files to get the smallest possible size.

	Please note that this utility is not able to read
	a generic file. It's granted to be able to read only
	the files generated by the AdvanceMAME emulator.

	To compress the files this utility uses the following
	strategies:

	* Remove all ancillary chunks.
	* Concatenate all the IDAT chunks.
	* Use the 7zip Deflate implementation.

Options
	-l, --list FILES...
		List the content of the specified files.

	-z, --recompress FILES...
		Recompress the specified files. If the -1, -2, -3
		options are specified it's used the smallest file
		choice from the previous compressed data and the
		new compression. If the -0 option is specified the
		file is always rewritten without any compression.

	-0, --shrink-store
		Disable the compression. The file is
		only stored and not compressed. The file is always
		rewritten also if it's bigger.

	-1, --shrink-fast
		Set the compression level to "fast" using the zlib
		compressor.

	-2, --shrink-normal
		Set the compression level to "normal" using the libdeflate
		compressor.
		This is the default level of compression.

	-3, --shrink-extra
		Set the compression level to "extra" using the 7z
		compressor.
		You can define the compressor iterations with
		the -i, --iter option.

	-4, --shrink-insane
		Set the compression level to "insane" using the zopfli
		compressor.
		You can define the compressor iterations with
		the -i, --iter option.

	-i, --iter N
		Define an additional numer of iterations for the 7z and zopfli
		compressors for modes -3 and -4.
		More iterations usually give a better compression, but may
		require a lot more time.
		Try for example with 10, 15, 20, and so on.

	-f, --force
		Force the use of the new file also if it's bigger.

Copyright
	This file is Copyright (C) 2002 Andrea Mazzoleni, Filipe Estima

See Also
	advzip(1), advmng(1), advdef(1)
